_search-plugins/knn/filter-search-knn.md

```diff
@@ -0,0 +1,650 @@
+{: .note }
+
+## How does a k-NN filter work?
+
+The OpenSearch k-NN plugin version 2.2 provided support for the Lucene engine to process k-NN searches. The Lucene engine provides a search that is based on the Hierarchical Navigable Small World (HSNW) algorithm to represent a multi-layered graph. The OpenSearch k-NN plugin version 2.4 is able to incorporate filters for searches based on Lucene 9.4.
+
+After a filter is applied to a set of documents to be searched, the algorithm decides whether to perform pre-filtering for an exact kNN search or modified post-filtering for approximate search. The approximate search with filtering guarantees the top number of closest vectors in the results.
+
+Lucene also provides the capability to operate its `KnnVectorQuery` over a subset of documents. To learn more about Lucene’s new capability, see the [Apache Lucene Documentation](https://issues.apache.org/jira/browse/LUCENE-10382).
+
+To learn more about all available k-NN search approaches, including approximate k-NN, exact k-NN with script score and pre-filtering with painless extensions, see [k-NN]({{site.url}}{{site.baseurl}}/opensearch/search-plugins/knn/index/).
+
+### Filtered search performance
+
+Filtering that is tightly integrated with the Lucene HNSW algorithm implementation allows you to to apply k-NN searches more efficiently, both in terms of relevancy of search results and performance. Consider, for example, an exact search using post-filtering on a large data set that returns results slowly and does not guarantee the required number of results specified by `k`.
+With this new capability, you can create an approximate k-NN search, apply filters, and get the number of results that you need. To learn more about approximate searches, see [Approximate k-nn search]({{site.url}}{{site.baseurl}}/opensearch/search-plugins/knn/approximate-knn/).
+
+The HSNW algorithm decides which type of filtering to apply to a search based on the volume of documents, and number of `k` points in the index that you search with a filter.
+
+![How the algorithm evaluates a doc set]({{site.url}}{{site.baseurl}}/images/hsnw-algorithm.png)
+
+| Variable | Description |
+-- | -- | -- |
+N | Number of documents in the index.
+P | Number of documents in the search set after the filter is applied using the formula: P <= N.
+q | The search vector.
+k | The maximum number of vectors to return in the response.
+
+To learn more about k-NN performance tuning, see [Performance tuning]({{site.url}}{{site.baseurl}}/opensearch/search-plugins/knn/performance-tuning/).
+
+## Filter approaches by use case
+
+Depending on the data set that you are searching, you might choose a different approach to minimize recall or latency. You can create filters that are either:
```

Comment:
Delete the word "either". If you keep it, the bullets need to be revised to use semicolons and the word "or" with the bullet preceding the last bullet. Also, lowercase the word "For".